rasa/core/evaluation/marker_base.py

```diff
@@ -578,44 +572,70 @@ def from_config(config: Any, name: Optional[Text] = None) -> Marker:
 
         return marker
 
-    def export_markers(
+    def evaluate_trackers(
         self,
-        tracker_loader: Iterator[Optional[DialogueStateTracker]],
-        output_file: Text,
-        stats_file: Optional[Text] = None,
+        trackers: Iterator[Optional[DialogueStateTracker]],
+        output_file: Path,
+        session_stats_file: Optional[Path] = None,
+        overall_stats_file: Optional[Path] = None,
     ) -> None:
         """Collect markers for each dialogue in each tracker loaded.
 
         Args:
-            tracker_loader: The tracker loader to use to select trackers for marker
-                            extraction.
+            trackers: An iterator over the trackers from which we want to extract
+                markers.
             output_file: Path to write out the extracted markers.
-            stats_file: (Optional) Path to write out statistics about the extracted
-                        markers.
-        """
-        processed_trackers = {}
+            session_stats_file: (Optional) Path to write out statistics about the
+                extracted markers for each session separately.
+            overall_stats_file: (Optional) Path to write out statistics about the
+                markers extracted from all session data.
 
-        for tracker in tracker_loader:
+        Raises:
+            `FileExistsError` if any of the specified files already exists
+            `NotADirectoryError` if any of the specified files is supposed to be
+                contained in a directory that does not exist
+        """
+        # Check files and folders before doing the costly swipe over the trackers:
+        for path in [session_stats_file, overall_stats_file, output_file]:
```

Comment:
I like the idea :D (had also put an overwrite parameter to the stats :D) - But since it wasn't in the list of requirements, let's just ignore it for now